### PR TITLE
Hunterize the harfbuzz cmakelists.txt

### PR DESCRIPTION
**Pull request by jhs67**
- add the HunterGate.cmake file
- include the hunter archive
- hunter_add_package for freetype and ICU
- install *Config.cmake files

### CODE REVIEW

File: cmake/harfbuzzConfig.cmake [hunk 1]
```diff
@@ -0,0 +1 @@
+include("${CMAKE_CURRENT_LIST_DIR}/harfbuzzTargets.cmake")
```
Comment (ruslo) [inline cmake/harfbuzzConfig.cmake]: `check_required_components` should be used here to check if user mistakenly specify some `COMPONENTS` in `find_package(harfbuzz ....)`:
* https://github.com/cgold-examples/fruits/blob/b266e5a8b0dccc250390346acde07e613562d85d/cmake/template/Config.cmake.in#L4

File: CMakeLists.txt [hunk 2]
```diff
@@ -785,6 +806,48 @@ if (NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
...
+  "${CMAKE_CURRENT_BINARY_DIR}/harfbuzz/harfbuzzConfigVersion.cmake"
+  VERSION ${HB_VERSION}
+  COMPATIBILITY AnyNewerVersion
+)
+
+install(TARGETS harfbuzz EXPORT harfbuzzTargets
+  ARCHIVE DESTINATION lib
+  LIBRARY DESTINATION lib
+  RUNTIME DESTINATION bin
+  INCLUDES DESTINATION include/harfbuzz
```
Comment (ruslo) [inline CMakeLists.txt]: Usually it's just `include`

File: cmake/harfbuzzConfig.cmake (hunk 1 above)
Comment (ruslo) [reply cmake/harfbuzzConfig.cmake]: Also we need `find_package` calls here for freetype/ICU dependencies when they used, example:
* https://github.com/elucideye/drishti/blob/9b11f52b8ce0fa93794f3afb7bcdbd0cfa518bc3/cmake/Config.cmake.in#L38-L49

**Review by ruslo — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (jhs67) [reply CMakeLists.txt]: `pkg-config harfbuzz --cflags` returns include/harfbuzz so I was mirroring that.

**Review by ruslo — APPROVED** [review]

**Comment by ruslo** [comment]
Released: https://github.com/hunter-packages/harfbuzz/releases/tag/v1.7.6-p0

**Pull request merged**